File: yumyum-back/src/main/java/com/yumyum/domain/feed/dto/FeedResponse.java
package com.yumyum.domain.feed.dto;

import com.yumyum.domain.feed.entity.Feed;
import com.yumyum.domain.map.entity.Place;
import com.yumyum.domain.user.entity.User;
import lombok.*;

import java.time.LocalDateTime;

@Data
@ToString
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class FeedResponse {

    private Long id;

    private String title;

    private String content;

    private Long score;

    private User user;

    private Place place;

    private String videoPath;

    private String thumbnailPath;

    private Boolean isCompleted;

    private Long likeCount;

    private Boolean isLike;

    private LocalDateTime createdDate;

    private LocalDateTime modifiedDate;

    public FeedResponse(final Feed feed, final User user, final Place place, final Long likeCount, final Boolean isLike){
        this.id = feed.getId();
        this.title = feed.getTitle();
        this.content = feed.getContent();
        this.score = feed.getScore();
        this.user = user;
        this.place = place;
        this.videoPath = feed.getVideoPath();
        this.thumbnailPath = feed.getThumbnailPath();
        this.isCompleted = feed.getIsCompleted();
        this.likeCount = likeCount;
        this.isLike = isLike;
        this.createdDate = feed.getCreatedDate();
        this.modifiedDate = feed.getModifiedDate();
    }
}

File: AOS/app/src/main/java/com/omnyom/yumyum/model/feed/FeedResponse.kt
package com.omnyom.yumyum.model.feed

import java.io.Serializable

data class FeedResponse(
    val `data`: List<FeedData>,
    val message: String,
    val status: String
)

data class SingleFeedResponse(
        val `data`: FeedData,
        val message: String,
        val status: String
)

data class FeedData(
        var content: String,
        val createdDate: String,
        val isCompleted: Boolean,
        val id: Int,
        var isLike: Boolean,
        var likeCount: Int,
        val modifiedDate: String,
        var place: Place?,
        val thumbnailPath: String,
        var title: String,
        val user: User,
        val videoPath: String,
        var score: Int
) : Serializable

data class User(
        val createdDate: String,
        val email: String,
        val id: Int,
        val introduction: String,
        val modifiedDate: String,
        val nickname: String,
        val profilePath: String
) : Serializable


data class Place(
        val address: String,
        val id: Int,
        val locationX: Double,
        val locationY: Double,
        val name: String,
        val phone: String
) : Serializable
